server/src/main/java/com/projects/pms/filter/JwtRequestFilter.java

```diff
@@ -0,0 +1,73 @@
+package com.projects.pms.filter;
+
+
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.projects.pms.services.EmployeeService;
+import com.projects.pms.util.JwtUtil;
+import io.jsonwebtoken.ExpiredJwtException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.web.authentication.WebAuthenticationDetailsSource;
+import org.springframework.stereotype.Component;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+@Component
+public class JwtRequestFilter extends OncePerRequestFilter {
+
+    @Autowired
+    private EmployeeService employeeService;
+
+    @Autowired
+    private JwtUtil jwtTokenUtil;
+
+    @Override
+    protected boolean shouldNotFilter(HttpServletRequest request) {
+        if(request.getRequestURI().equalsIgnoreCase("/login")){
+            return true;
+        }
+        else{
+            return false;
+        }
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request, HttpServletResponse response, FilterChain chain)
+            throws ServletException, IOException {
+
+        final String requestTokenHeader = request.getHeader("Authorization");
```

Comment:
@SunilSunny698 Why as an final, is it required ?

Reply:
It is not required but we thought it is a good practice to mention it as a final variable as we are not going to override its value anywhere.